### PR TITLE
User crud 

### PR DESCRIPTION
**Pull request by Usama-Khaan**
Add a new user. - An email should be sent to the user being added, through which that user will be able to sign up to our application.
Edit an existing user.
Delete a user.
List all users.

### CODE REVIEW

File: app/controllers/users/registrations_controller.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+class Users::RegistrationsController < Devise::RegistrationsController
+  before_action :configure_sign_up_params, only: [:create]
+  before_action :configure_account_update_params, only: [:update]
+
+  # If you have extra params to permit, append them to the sanitizer.
+  def configure_sign_up_params
+    devise_parameter_sanitizer.permit(:sign_up, keys: [:username])
+  end
+
+  # If you have extra params to permit, append them to the sanitizer.
```
Comment (burraqxprolabs) [inline app/controllers/users/registrations_controller.rb]: Please remove comments

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
+class UsersController < ApplicationController
```
Comment (burraqxprolabs) [inline app/controllers/users_controller.rb:1]: Add one empty line after class code

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+  def destroy
+    @user = User.find(params[:id])
+    @user.destroy
+    redirect_to root_path, status: :see_other
+  end
+
+  def user_params
+    params.require(:user).permit(:email, :username)
+  end
```
Comment (burraqxprolabs) [inline app/controllers/users_controller.rb:27]: Add one empty line always last end before

File: app/helpers/users_helper.rb [hunk 4]
```diff
@@ -0,0 +1,2 @@
+module UsersHelper
```
Comment (burraqxprolabs) [inline app/helpers/users_helper.rb]: If we're not using this helper, we don't need to commit it.

File: app/models/user.rb [hunk 5]
```diff
@@ -0,0 +1,8 @@
+class User < ApplicationRecord
+  # Include default devise modules. Others available are:
```
Comment (burraqxprolabs) [inline app/models/user.rb]: Please remove commented code

File: app/views/users/_form.html.erb [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+    <%= form.text_area :email %><br>
+    <% user.errors.full_messages_for(:body).each do |message| %>
+      <div><%= message %></div>
+    <% end %>
+  </div>
+
+  <div>
+    <%= form.submit %>
+  </div>
+<% end %>
```
Comment (burraqxprolabs) [inline app/views/users/_form.html.erb]: Always add empty line end to the end of file

File: app/views/users/edit.html.erb [hunk 7]
```diff
@@ -0,0 +1,2 @@
+<h1>Edit user</h1>
+<%= render "form", user: @user%>
```
Comment (burraqxprolabs) [inline app/views/users/edit.html.erb]: Here also add empty line

File: test/controllers/users_controller_test.rb [hunk 8]
```diff
@@ -0,0 +1,7 @@
+require "test_helper"
```
Comment (burraqxprolabs) [inline test/controllers/users_controller_test.rb]: We don't need this file in this commit

File: app/views/users/edit.html.erb (hunk 7 above)
Comment (Usama-Khaan) [inline app/views/users/edit.html.erb]: space

**Pull request merged**